app/views/components/warning.scala.html

```diff
@@ -14,15 +14,11 @@
  * limitations under the License.
  *@
 
-@this()
+@this(govukWarningText: GovukWarningText)
 
 @(content: Html)(implicit messages: Messages)
 
-<div class="govuk-warning-text">
- <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
- <strong class="govuk-warning-text__text">
-  <span class="govuk-warning-text__assistive">@messages("site.warning")</span>
-  @content
- </strong>
-</div>
-
+@govukWarningText(WarningText(
+  iconFallbackText = Some(messages("site.warning")),
+  content = HtmlContent(content)
```

Comment:
Updated here to use `GovukWarningText` to fix issue with warning text and used `HtmlContent` instead of `Text` as Html is passed here and didn't want to change much in the codebase.